java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -0,0 +1,487 @@
+  }
+
+  private void handleBranch(CFG.Block programPosition, Tree condition, boolean checkPath) {
+    Pair<List<ProgramState>, List<ProgramState>> pair = constraintManager.assumeDual(programState);
+    for (ProgramState state : pair.a) {
+      // enqueue false-branch, if feasible
+      ProgramState ps = ProgramState.stackValue(state, SymbolicValue.FALSE_LITERAL);
+      enqueue(new ExplodedGraph.ProgramPoint(programPosition.falseBlock(), 0), ps);
+      if (checkPath) {
+        alwaysTrueOrFalseChecker.evaluatedToFalse(condition);
+      }
+    }
+    for (ProgramState state : pair.b) {
+      ProgramState ps = ProgramState.stackValue(state, SymbolicValue.TRUE_LITERAL);
+      // enqueue true-branch, if feasible
+      enqueue(new ExplodedGraph.ProgramPoint(programPosition.trueBlock(), 0), ps);
+      if (checkPath) {
+        alwaysTrueOrFalseChecker.evaluatedToTrue(condition);
+      }
+    }
+  }
+
+  private void visit(Tree tree, Tree terminator) {
+    LOG.debug("visiting node " + tree.kind().name() + " at line " + ((JavaTree) tree).getLine());
+    if (!checkerDispatcher.executeCheckPreStatement(tree)) {
+      // Some of the check pre statement sink the execution on this node.
+      return;
+    }
+    switch (tree.kind()) {
+      case METHOD_INVOCATION:
+        MethodInvocationTree mit = (MethodInvocationTree) tree;
+        setSymbolicValueOnFields(mit);
+        //unstack arguments and method identifier
+        programState = ProgramState.unstack(programState, mit.arguments().size() + 1).a;
+        logState(mit);
+        programState = ProgramState.stackValue(programState, constraintManager.createSymbolicValue(mit));
+        break;
+      case LABELED_STATEMENT:
+      case SWITCH_STATEMENT:
+      case EXPRESSION_STATEMENT:
+      case PARENTHESIZED_EXPRESSION:
+        throw new IllegalStateException("Cannot appear in CFG: " + tree.kind().name());
+      case VARIABLE:
+        VariableTree variableTree = (VariableTree) tree;
+        ExpressionTree initializer = variableTree.initializer();
+        if (initializer == null) {
+          SymbolicValue sv = null;
+          if (terminator != null && terminator.is(Tree.Kind.FOR_EACH_STATEMENT)) {
+            sv = constraintManager.createSymbolicValue(variableTree);
+          } else if (variableTree.type().symbolType().is("boolean")) {
+            sv = SymbolicValue.FALSE_LITERAL;
+          } else if (!variableTree.type().symbolType().isPrimitive()) {
+            sv = SymbolicValue.NULL_LITERAL;
+          }
+          if (sv != null) {
+            programState = ProgramState.put(programState, variableTree.symbol(), sv);
+          }
+        } else {
+          Pair<ProgramState, List<SymbolicValue>> unstack = ProgramState.unstack(programState, 1);
+          programState = unstack.a;
+          programState = ProgramState.put(programState, variableTree.symbol(), unstack.b.get(0));
+        }
+        break;
+      case TYPE_CAST:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 8 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1151)